### PR TITLE
[Synapse] Update version of reference package to use stable api version

### PR DESCRIPTION
**Pull request by wonner**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Synapse/Synapse/Models/SynapseAnalyticsSparkClient.cs [hunk 1]
```diff
@@ -193,24 +183,13 @@ public void ResetSparkSessionTimeout(int sessionId)
...
-        {
-            return Poll(
-                statement,
-                s => null,
-                s => s.State,
-                s => this.GetSparkSessionStatement(sessionId, s.Id),
-                SparkSessionStatementLivyState.ExecutingStates,
-                isFinalState: false);
+            return statement.Poll().Value;
         }
```
Comment (idear1203) [inline src/Synapse/Synapse/Models/SynapseAnalyticsSparkClient.cs:193]: Please double confirm the code change here is validate

**Comment by idear1203** [comment]
@wonner I see this PR also changes the version of Azure.Core. Could you please work with Azure PowerShell team to check if the change looks good.

**Review by idear1203 — APPROVED** [review]

**Review by idear1203 — DISMISSED** [review]

**Comment by wyunchi-ms** [comment]
Please resolve the conflicts.

**Comment by wonner** [comment]
@wyunchi-ms Conflicts have been resolved. Could you please help review again?

**Comment by wyunchi-ms** [comment]
Please resolve the conflicts. Storage team has updated the version of `Azure.Core`. Could you please revert this change in your PR?

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**